File: Arrays _101/1346. Check If N and Its Double Exist/check.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        mp = {}
        for item in arr:
            if item not in mp.keys():
                mp[item] = 1
            else:
                mp[item] = mp[item] + 1
        for i, item in enumerate(arr):
            if item in mp.keys():
                if item == 0 and mp[0] == 1:
                    continue
                if 2 * item in mp.keys():
                    return True
        return False

File: Arrays 101/1346. Check If N and Its Double Exist/check.cpp
class Solution
{
public:
    bool checkIfExist(vector<int> &arr)
    {
        unordered_map<int, int> mp;
        for (auto it : arr)
        {
            mp[it]++;
        }

        for (int i = 0; i < arr.size(); i++)
        {
            if (arr[i] == 0 && mp[0] == 1)
            {
                continue;
            }
            if (mp.find(2 * arr[i]) != mp.end())
            {
                return true;
            }
        }
        return false;
    }
};